### DISCUSSION TITLE
Running two Renovate Jobs in one Pipeline with different Settings causing in closed PR's

### DISCUSSION THREAD

**Discussion by schamal2205** [renovatebot/renovate#18308] [Request Help] [answered]
To match our use-case i tried to configure Renovate within one Pipeline to use different config.js files. 
The first one should update our internal Packages with major, minor and patch version in one PullRequest and also should not ignore unstable Packages. To realize this i've created the following Pipeline YAML: 
```YAML
steps:
  - bash: |
      > config.js

      echo "module.exports = {" >> config.js
      echo "  platform: 'azure'," >> config.js
      echo "  endpoint: 'https://dev.azure.com/<company>/'," >> config.js
      echo "  token: process.env.TOKEN," >> config.js
      echo "  nuget: {" >> config.js
      echo "    registryUrls: [" >> config.js
      echo "      'https://pkgs.dev.azure.com/<company>/_packaging/<project>/nuget/v3/index.json'" >> config.js
      echo "    ]" >> config.js
      echo "  }," >> config.js
      echo "  packageRules: [" >> config.js
      echo "    {" >> config.js
      echo "      groupName: 'RootNine (all)'," >> config.js
      echo "      matchPackagePatterns: ['^RootNine']," >> config.js
      echo "    }" >> config.js
      echo "  ]," >> config.js
      echo "  hostRules: [" >> config.js
      echo "    {" >> config.js
      echo "      hostType: 'nuget'," >> config.js
      echo "      matchHost: 'https://pkgs.dev.azure.com'," >> config.js
      echo "      username: ''," >> config.js
      echo "      password: '<password>'," >> config.js
      echo "    }," >> config.js
      echo "  ]," >> config.js
      echo "  repositories: ['<repo>']," >> config.js
      echo "};" >> config.js

  - bash: |
      git config --global user.email 'bot@renovateapp.com'
      git config --global user.name 'Renovate Bot'
      npx renovate
    env:
      TOKEN: $(System.AccessToken)
      RENOVATE_CONFIG_FILE: "config.js"
      LOG_LEVEL: "debug"
      RENOVATE_BASE_BRANCHES: "$(Build.SourceBranchName)"
      RENOVATE_IGNORE_UNSTABLE: "false"
      RENOVATE_SEPARATE_MAJOR_MINOR: "false"
```
The second one (running in the same Pipeline afterwards) should update all external packages with minor and patch updates in one Pullrequest and major updates in an extra PR. 
```YAML
  - bash: |
      > config.js

      echo "module.exports = {" >> config.js
      echo "  platform: 'azure'," >> config.js
      echo "  endpoint: 'https://dev.azure.com/<company>/'," >> config.js
      echo "  token: process.env.TOKEN," >> config.js
      echo "  nuget: {" >> config.js
      echo "    registryUrls: [" >> config.js
      echo "      'https://api.nuget.org/v3/index.json'," >> config.js
      echo "    ]" >> config.js
      echo "  }," >> config.js
      echo "  packageRules: [" >> config.js
      echo "    {" >> config.js
      echo "      groupName: 'External (minor, patch)'," >> config.js
      echo "      excludePackagePatterns: ['^RootNine']," >> config.js
      echo "    }" >> config.js
      echo "  ]," >> config.js
      echo "  hostRules: [" >> config.js
      echo "    {" >> config.js
      echo "      hostType: 'nuget'," >> config.js
      echo "      matchHost: 'https://pkgs.dev.azure.com'," >> config.js
      echo "      username: ''," >> config.js
      echo "      password: '<password>' >> config.js
      echo "    }," >> config.js
      echo "  ]," >> config.js
      echo "  repositories: ['<repo>']," >> config.js
      echo "};" >> config.js

  - bash: |
      git config --global user.email 'bot@renovateapp.com'
      git config --global user.name 'Renovate Bot'
      npx renovate
    env:
      TOKEN: $(System.AccessToken)
      RENOVATE_CONFIG_FILE: "config.js"
      LOG_LEVEL: "debug"
      RENOVATE_BASE_BRANCHES: "$(Build.SourceBranchName)"
      RENOVATE_IGNORE_UNSTABLE: "true"
      RENOVATE_SEPARATE_MAJOR_MINOR: "true"
      GITHUB_COM_TOKEN: "<GH-Token>"
```

The functionality is more or less working the first one opens a pull-request with with all the external packages. But when the second one is running and the PR's for the external Packages are opened the first one for the internal gets automatically abandoned. 

So what I want to achieve is: 
- Open PR for new internal Packages 
- If there are newer internal Packages open a new PR and close the First one 
- Open PR for (minor, patch) external Packages and leave the internal PR's open 
- If there are newer external Packages open a new PR and close the older one 
- Open PR for major external Packages 
- If there are newer external major Packages open a new PR and close the older one 

Is there any way to run or configure renovate to match my usecase?

We would really want to change our dependency automation from NuKeeper/Dependabot to Renovate but getting this not configured is currently blocking the replacement :(.

**Comment by viceice** [comment]
don't run different configs on one repo.

use a single config with appropriate package rules instead.


otherwise it's very complicated to get it working and we strongly discourage that. it's official not supported.

**Comment by rarkins** [comment]
You can set [pruneStaleBranches](https://docs.renovatebot.com/configuration-options/#prunestalebranches) to `false` but like @viceice I recommend against this type of approach and instead you should work out what's blocking you from using a single config

**Reply by schamal2205** [reply, marked as answer]
Perfect thank you. I've figured out how to make the configuration whitin one config file.
